### PR TITLE
Refactor value types

### PR DESCRIPTION
**Pull request by sdlyy**
Resolves L2B-2083

## What's changed
- Removed `ValueType`
- Introduced `AggregatedReportType` & `ReportType`
- `AssetType` is already present for tokens
- Refactored interfaces respectively
- Refactored & replaced backend & frontend code
- Added renaming migrations 

## Notes
Deploy and build will fail since staging exposes old API format

### CODE REVIEW

**Comment by sdlyy** [comment]
Code changes are ready for review, in the meantime I gotta double check migrations and stuff

File: packages/config/src/layer2s/arbitrum.ts [hunk 1]
```diff
@@ -102,7 +101,7 @@ const TOKENS: Omit<Token, 'chainId'>[] = [
     decimals: 6,
     category: 'stablecoin',
     premintHolderAddresses: [],
-    type: ValueType.EBV,
+    type: 'NMV',
```
Comment (antooni) [inline packages/config/src/layer2s/arbitrum.ts:104]: typo

File: packages/backend/src/peripherals/database/migrations/070_update_usdc_id.ts [hunk 2]
```diff
@@ -22,23 +22,27 @@ export async function up(knex: Knex) {
...
-    project_id: ProjectId.ARBITRUM.toString(),
-  })
+  await knex('reports')
+    .update({ asset_id: NEW_ASSET_ID })
+    .where({
+      chain_id: +ChainId.ARBITRUM,
+      asset_id: OLD_ASSET_ID,
+      project_id: ProjectId.ARBITRUM.toString(),
+    })
+    .andWhereRaw(`"asset_type" = 'EBV'`)
```
Comment (antooni) [inline packages/backend/src/peripherals/database/migrations/070_update_usdc_id.ts:32]: are those double quotes aroung asset_type needed?

**Review by antooni — COMMENTED** [review]
looks good, left small comments

File: packages/config/src/layer2s/arbitrum.ts (hunk 1 above)
Comment (sdlyy) [reply packages/config/src/layer2s/arbitrum.ts]: https://github.com/l2beat/l2beat/pull/1807/commits/9dc7444e9e50cc5489aa655d55d52319dc02d2ee

File: packages/backend/src/peripherals/database/AggregatedReportRepository.ts [hunk 3]
```diff
@@ -147,6 +154,7 @@ function toRecord(row: AggregatedReportRow): AggregatedReportRecord {
     projectId: ProjectId(row.project_id),
     usdValue: BigInt(row.usd_value),
     ethValue: BigInt(row.eth_value),
-    valueType: ValueType(row.value_type),
+    // TODO: rename this
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/peripherals/database/AggregatedReportRepository.ts]: https://github.com/l2beat/l2beat/pull/1807/commits/0c2371dae27d9b205d3d09f4d48c5d90e3fc938f

File: packages/backend/src/peripherals/database/ReportRepository.ts [hunk 4]
```diff
@@ -19,7 +19,8 @@ export interface ReportRecord {
   asset: AssetId
   chainId: ChainId
   // TODO: Index this column when we start querying by it.
-  type: ValueType
+  // TODO: Rename
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/peripherals/database/ReportRepository.ts:22]: https://github.com/l2beat/l2beat/pull/1807/commits/2bd6dde2b16b51d06c69991a3f854ca5a4676fcc

**Review by antooni — APPROVED** [review]

**Review by antooni — APPROVED** [review]

**Pull request merged**